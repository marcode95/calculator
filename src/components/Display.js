import React from 'react';
import PropTypes from 'prop-types';

const Display = (props) => { // eslint-disable-line
  return (
    <>
      <div>{props.result}</div> {/*eslint-disable-line*/}
    </>
  );
};

Display.propTypes = {
  result: PropTypes.string, // eslint-disable-line
};
Display.defaultProbs = {
  result: '0',
};

export default Display;
