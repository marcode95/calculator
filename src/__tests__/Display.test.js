import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Display from '../components/Display';
import '@testing-library/jest-dom/extend-expect';

describe('checks if Display renders', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render right value', () => {
    const total = '23';

    render(<Display total={total} />);
    const text = screen.getByText('23');
    expect(text).toBeInTheDocument();
  });
});