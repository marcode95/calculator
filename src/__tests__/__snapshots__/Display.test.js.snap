// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Display renders should render correctly 1`] = `
<div
  className="display"
>
  <p
    className="result"
    style={
      Object {
        "display": "block",
      }
    }
  >
    0
  </p>
  <p
    className="operation"
  />
  <p
    className="next"
  />
</div>
`;
